bindings/ocaml/Hacl.ml

```diff
@@ -293,30 +290,47 @@ module NaCl = struct
         get_result @@ hacl_NaCl_crypto_secretbox_open_easy (C.ctypes_buf pt) (C.ctypes_buf ct) (C.size_uint32 ct) (C.ctypes_buf n) (C.ctypes_buf key)
     end
     module Detached = struct
-      let box ~pt ~n ~pk ~sk ~ct ~tag =
+      let get_sub_buffer buf offset len =
```

Comment:
Usually, you need to check for `0 <= offset && 0 <= len && len <= C.size buf - offset`, as, for example, `offset + len` or `C.size buf - offset` may overflow in some cases.

Reply:
Thanks, updated